superset/exceptions.py

```diff
@@ -37,7 +37,19 @@ def exception(self) -> Optional[Exception]:
 
 
 class SupersetTimeoutException(SupersetException):
-    pass
+    status = 408
+
+    def __init__(
+        self,
+        error_type: SupersetErrorType,
+        message: str,
+        level: ErrorLevel,
+        extra: Optional[Dict[str, Any]],
+    ) -> None:
+        super(SupersetTimeoutException, self).__init__(message)
+        self.error = SupersetError(
+            error_type=error_type, message=message, level=level, extra=extra
+        )
```

Comment:
I'm wondering whether we can just extend the attributes of `SupersetError` directly on `SupersetException` and use a custom JSON encoder to serialize it. This way you remove a layer of abstraction and the need to map the arguments for any new errors your might add in the future,

Reply:
Hmm, possibly. I'm not super familiar with what the plan is for exception handling on the Python side, but having the SupersetExceptions automatically contain the error properties might make sense.

The reason I shied away from it originally was so that it was very clear what attributes the clients need to render the errors and what parts are server specific. But maybe that's not a big issue. Maybe @john-bodley has thoughts?

Reply:
@ktmud could you expand (possibly with an example) on your thoughts? Are you looking for more of a generic container for handling error properties?

Reply:
I was thinking maybe subclasses of `SupersetException` itself can handle encoding of error code and messages. Here's an example:

```python
class SupersetException(Exception, SupersetAlert):
    def __init__(...) -> None:
        SupersetAlert.__init__(
            self, code=code, message=message, description=description, extra=extra
        )
        Exception.__init__(self, self.message)

def json_error_response(error: SupersetException) -> FlaskResponse:
   return jsonify({ "error": error.to_json() })
```

[Full gist here](https://gist.github.com/ktmud/59ccbdf6c96570e1ffe41ee8e5ecd479)